packages/runtime/runtime-utils/src/summarizerNodeWithReferences.ts

```diff
@@ -0,0 +1,242 @@
+/*!
```

Comment:
That's a lot of new code, and no UTs? :)

Reply:
The only new code is adding the fluid object references. Rest if duplicate of SummarizerNode. I updated dataStoreContext.spec.ts and dataStoreCreation.spec.ts to use this new class and it worked fine.
I will add some tests that verify we get correct references.

Reply:
Added UTs for the added functionality.